client/components/parts/ImageSearchField.jsx

```diff
@@ -28,63 +27,64 @@ export default class ImageSearchField extends Component {
   }
 
   handleChange = (e) => {
-    this.props.updateImage(e.target.id)
+    this.props.updateImage(e.target.dataset['url'])
   }
 
-  searchByKeyword = () => {
-    const unsplashClientId = 'd9ac3268ec6e896a2f2655d25b135b0d22dab5762719fbe4fbc25864860ab8e0'
-    // const unsplashClientId = 'ef9e27a50e4d36d08c24bb35f362974c36420854e6924ad0e4b79c4c6dd8b041'
-    const unsplashSearchByKeywordUrl = `https://api.unsplash.com/search/photos/?client_id=${unsplashClientId}&query=${this.state.searchKeywords}`
-    const request = new Request(
-      unsplashSearchByKeywordUrl,
-      {method: 'GET'}
-    )
-    return Promise.resolve(fetch(request).then((response) => {
-      return response.json().then((content) => {
-        return (content.results).map((imgObject) => {
-          return {
-            'thumbnail': imgObject.urls.thumb,
-            'full': imgObject.urls.full
-          }
-        })
+
+  // Clubhub Client ID: d9ac3268ec6e896a2f2655d25b135b0d22dab5762719fbe4fbc25864860ab8e0
+  // Juan's  Client ID: ef9e27a50e4d36d08c24bb35f362974c36420854e6924ad0e4b79c4c6dd8b041
+
+  searchByKeyword = () => new Promise((resolve, reject) => {
+    clearTimeout(this.searchTimer)
+    this.searchTimer = setTimeout(() => {
+      // Send search request
+      fetch(`https://api.unsplash.com/search/photos/?client_id=${Config.unsplash_client_id}&query=${this.state.searchKeywords}`)
+      .then((response) => {
+        if (response.ok){
+          resolve(response.json().then((json) => (json.results).map((imgObject) =>
+            ({
+              'thumbnail': imgObject.urls.thumb,
+              'full': imgObject.urls.full
+            })
+          )))
+        } else {
+          reject(response.status)
+        }
       })
-    }))
-  }
+    }, 500) // <- Wait 500ms before querying
```

Comment:
Yay timeouts! 
The global this.searchTimer hurts my heart a bit but it's worth it